core/src/main/java/engineer/thesis/core/service/TimeSlotService.java

```diff
@@ -0,0 +1,64 @@
+package engineer.thesis.core.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import engineer.thesis.core.model.Doctor;
+import engineer.thesis.core.model.TimeSlot;
+import engineer.thesis.core.model.dto.TimeSlotDTO;
+import engineer.thesis.core.repository.DoctorRepository;
+import engineer.thesis.core.repository.TimeSlotRepository;
+
+@Service
+public class TimeSlotService implements ITimeSlotService {
+
+	@Autowired
+	private TimeSlotRepository timeSlotRepository;
+
+	@Autowired
+	private DoctorRepository doctorRepository;
+
+	@Override
+	public TimeSlotDTO saveTimeSlot(TimeSlotDTO timeSlotDTO, long doctorId) throws IllegalArgumentException {
+		if (timeSlotRepository.exists(timeSlotDTO.getId()) && !isOwner(doctorId, timeSlotDTO.getId())) {
```

Comment:
Jakby to był poważny projekt z zapasem czasu to bym się czepiał że do bazy leci więcej zapytań niż trzeba ale na nasz potrzeby jest git